File: code.leetcode/lianbiao/easy/e83.删除排序链表中的重复元素.cpp
#include <unordered_map>
struct ListNode
{
    int val;
    ListNode *next;
    ListNode() : val(0), next(nullptr) {}
    ListNode(int x) : val(x), next(nullptr) {}
    ListNode(int x, ListNode *next) : val(x), next(next) {}
};
class Solution
{
public:
    ListNode *deleteDuplicates(ListNode *head)
    {
        // 遍历链表
        // 申请ya节点
        ListNode* preHead = new ListNode(0);
        preHead->next = head;
        
        ListNode *ya = preHead;
        // 申请map
        unordered_map<int, bool> m = {} while (ya->next != nullptr)
        {

            if (m[ya->next->val])
            {
                ya->next = ya->next->next;
            }
            else
            {

                m[ya->next->val] = true;
                ya = ya->next;
            }
        }
        // map记录
        // 若已存在，删除
        return preHead->next;
    }
};

File: code.leetcode/lianbiao/easy/e83.删除排序链表中的重复元素.go
package lianbiao

func deleteDuplicates(head *ListNode) *ListNode {
	//遍历链表
	//申请ya节点
	dummy := &ListNode{Val: 0, Next: head}
	ya := dummy
	//申请map
	m := make(map[int]bool)
	for ya.Next != nil {

		if m[ya.Next.Val] {
			ya.Next = ya.Next.Next
		} else {

			m[ya.Next.Val] = true
			ya = ya.Next
		}

	}
	//map记录
	//若已存在，删除
	return dummy.Next
}
